keras/engine/training.py

```diff
@@ -258,7 +258,10 @@ def collect_trainable_weights(layer):
         weights += layer.trainable_weights
     # dedupe weights
     weights = list(set(weights))
-    weights.sort(key=lambda x: x.name)
+    try:
+        weights.sort(key=lambda x: x.name)
+    except TypeError:
+        pass
```

Comment:
So in Python 3 the sort method throws a TypeError, when the `name` is `None`. This doesn't occur in Python 2, so one has to catch this exception.